model_compression_toolkit/keras/kpi_data_facade.py

```diff
@@ -0,0 +1,98 @@
+from model_compression_toolkit.common.framework_info import FrameworkInfo
+from model_compression_toolkit.common.mixed_precision.mixed_precision_quantization_config import \
+    DEFAULT_MIXEDPRECISION_CONFIG
+from model_compression_toolkit.keras.constants import DEFAULT_HWM
+
+import importlib
+
+
+if importlib.util.find_spec("tensorflow") is not None\
+        and importlib.util.find_spec("tensorflow_model_optimization") is not None:
+    from model_compression_toolkit.keras.default_framework_info import DEFAULT_KERAS_INFO
+    from model_compression_toolkit.keras.keras_implementation import KerasImplementation
+    from tensorflow.keras.models import Model
+
+    from model_compression_toolkit import get_model
+
+    KERAS_DEFAULT_MODEL = get_model(TENSORFLOW, DEFAULT_HWM)
+
+
+    def keras_kpi_data(in_model: Model,
+                       representative_data_gen: Callable,
+                       quant_config: MixedPrecisionQuantizationConfig = DEFAULT_MIXEDPRECISION_CONFIG,
+                       fw_info: FrameworkInfo = DEFAULT_KERAS_INFO,
+                       fw_hw_model: FrameworkHardwareModel = KERAS_DEFAULT_MODEL) -> KPI:
+        """
+        Computes KPI data that can be used to calculate the desired target KPI for mixed-precision quantization.
+        Builds the computation graph from the given model and hw modeling, and uses it to compute the KPI data.
+
+        Args:
+            in_model (Model): Keras model to quantize.
+            representative_data_gen (Callable): Dataset used for calibration.
+            quant_config (MixedPrecisionQuantizationConfig): MixedPrecisionQuantizationConfig containing parameters of how the model should be quantized.
+            fw_info (FrameworkInfo): Information needed for quantization about the specific framework (e.g., kernel channels indices, groups of layers by how they should be quantized, etc.). `Default Keras info <https://github.com/sony/model_optimization/blob/21e21c95ca25a31874a5be7af9dd2dd5da8f3a10/model_compression_toolkit/keras/default_framework_info.py#L113>`_
+            fw_hw_model (FrameworkHardwareModel): FrameworkHardwareModel to optimize the Keras model according to.
+
+        Returns:
+            A KPI object with total weights parameters sum and max activation tensor.
+
+        Examples:
+            Import a Keras model:
+
+            >>> from tensorflow.keras.applications.mobilenet import MobileNet
+            >>> model = MobileNet()
+
+            Create a random dataset generator:
+
+            >>> import numpy as np
+            >>> def repr_datagen(): return [np.random.random((1,224,224,3))]
+
+            TODO: what is the correct way to explain about how to provide fw hw model? (do we need to give example for this if it is not mandatory?)
+            Import mct and call for KPI data calculation:
+            >>> import model_compression_toolkit as mct
+            >>> kpi_data_dict = keras_kpi_data(model, repr_datagen)
+
```

Comment:
No need to explain here about HW model. Only mandatory arguments. But let's write in the function documentation about the default values. Something like: https://github.com/sony/model_optimization/blob/main/model_compression_toolkit/keras/quantization_facade.py#L117

Reply:
And I would rename kpi_data_dict to kpi_data as this is not a dictionary.

Reply:
It was a dict initially and I forgot to change the name, fixed

Reply:
> No need to explain here about HW model. Only mandatory arguments. But let's write in the function documentation about the default values. Something like: https://github.com/sony/model_optimization/blob/main/model_compression_toolkit/keras/quantization_facade.py#L117

I already added a documentation about the default value for fw_info, which is the one that has default value in the quantization_facade (the link you added).
Do you mean we should add default to other arguments' documentation as well? which ones?